### DISCUSSION TITLE
Handle Navigation to another route to prevent data loss

### DISCUSSION THREAD

**Discussion by jangwalia** [vercel/next.js#61060] [Help]
### Summary

I am working on a project which is built in next.js@13 with app directory, The project contains a form which user filled and if user wants to leave the page it should ask for permission , but I couldn't find a way to do that with usePathname or useRouter from the next js 13 , there are some examples which use SingletonRouter form 'next/router' but it is deprecated now, can anyone please help me with this
Thanks :)

### Additional information

```js
This is the reference I am using https://medium.com/better-programming/prevent-route-changes-and-unsaved-data-loss-in-next-js-f93622d73791
```

**Comment by seblondono** [comment]
This is [a comment](https://github.com/vercel/next.js/discussions/49532#discussioncomment-9081115) I made on a related issue. TL;DR persists user's unfinished form state on session storage instead of preventing route transition. Better UX, smaller tech effort.
